worker/upload.go

```diff
@@ -54,18 +55,21 @@ type (
 	}
 
 	uploader struct {
-		host       hostV3
-		fcid       types.FileContractID
+		mgr *uploadManager
+
 		hk         types.PublicKey
 		siamuxAddr string
-		endHeight  uint64
 
 		statsSectorUploadEstimateInMS    *dataPoints
 		statsSectorUploadSpeedBytesPerMS *dataPoints // keep track of this separately for stats (no decay is applied)
 		signalNewUpload                  chan struct{}
 		stopChan                         chan struct{}
 
 		mu                  sync.Mutex
+		host                hostV3
+		fcid                types.FileContractID
+		renewedFrom         types.FileContractID
```

Comment:
How about giving every uploader a random id? To avoid `renewedFrom`? Because it seems like it's only used for the `used` check. A random id would always stay the same while still allowing for multiple contracts per host.